src/Automatonymous/AutomatonymousStateMachine.cs

```diff
@@ -1185,5 +1358,23 @@ public void Declare(object stateMachine)
                 }
             }
         }
+
+        private StateMachine<TInstance> Modify(Action<StateMachineModifier<TInstance>> modifier)
+        {
+            StateMachineModifier<TInstance> builder = new InternalStateMachineModifier<TInstance>(this);
+            modifier(builder);
+            builder.Apply();
+
+            return this;
+        }
+
+        private class BuilderStateMachine : AutomatonymousStateMachine<TInstance> { }
+
+        public static AutomatonymousStateMachine<TInstance> Build(Action<StateMachineModifier<TInstance>> modifier)

```

Comment:
I'm not sure I like this anchor, it feels like it should be elsewhere to align with the factory method approach used by all the MassTransit related project. Or maybe this is a sensible place, I can't think of another anchor and having a non-generic version of ASM would be annoying.

Maybe change Build to Create() or CreateUsingBuilder()

Reply:
Funny, I thought about `Create` first too. If you think that is appropriate, I'll rename it. As for a better anchor, I played with a couple different approaches including opening the builder up publicly to allow it directly, but it felt like too big of a change. However, creating a static class for `AutomatonymousStateMachine` with a single `Create<TInstance>` wouldn't be hard to do and might be better. How do you feel about this:

```csharp
public static class AutomatonymousStateMachine
{
  public static StateMachine<TInstance> Create<TInstance>(StateMachineModifier<TInstance> builder)
  {
    var machine = new BuilderStateMachine();
    machine.Modify(modifier);
    return machine;
  }

  private class BuilderStateMachine : AutomatonymousStateMachine<TInstance> { }
}
```

I can move the internal BuilderStateMachine here as well. I wouldn't inherit it from the current one. It would simply be a helper static class but externally wouldn't require any generic on the class itself. Just a thought.